### DISCUSSION TITLE
Convenience functions for database models to load/reload a specific field.

### DISCUSSION THREAD

**Discussion by orible** [aarondl/sqlboiler#929] [Ideas]
Would it be possible/ a good idea to add convenience functions to database models to load/reload a specific field.

For example
`
type Post {
UserID uint
Text string
}
`

(an already loaded model)
p := &model.Post{}
....
err := p.LoadText(ctx, db)

I only raise the question as it occurred to me that I can either p.Reload(ctx, db) or run model.Posts(...) for the specific field, but it seems slightly redundant if I'm dealing with an already loaded object.

**Comment by aarondl** [comment]
Does it make sense to add a list of fields to reload as an optional part of p.Reload()?

**Reply by orible** [reply]
Yes, I would say so 🤔 

That would allow runtime reloading of arbitrary fields in an easy way
`
var fields []string

fields = append(fields, model.PostsColumns.Text)

p.ReloadColumns(ctx, db, fields...)
`

But that does also make it slightly verbose to reload a single field
p.ReloadColumns(ctx, db, []string{model.PostColumns.Text})

**Reply by aarondl** [reply]
Generally I'd recommend a `...string` for this sort of thing, else we would not be able to do it as it'd be a breaking change.

**Reply by orible** [reply]
Aye, makes sense
